mypy/fastparse2.py

```diff
@@ -586,15 +586,15 @@ def try_handler(self,
                     orelse: List[ast27.stmt],
                     finalbody: List[ast27.stmt],
                     lineno: int) -> TryStmt:
-        def produce_name(item: ast27.ExceptHandler) -> Optional[NameExpr]:
+        vs = []
+        for item in handlers:
             if item.name is None:
-                return None
+                vs.append(None)
             elif isinstance(item.name, ast27.Name):
-                return NameExpr(item.name.id)
+                vs.append(NameExpr(item.name.id))
             else:
-                raise RuntimeError("'{}' has non-Name name.".format(ast27.dump(item)))
-
-        vs = [produce_name(h) for h in handlers]
+                self.fail("Sorry, `except <expr>, (<tuple>)` is not supported", item.lineno, item.col_offset)
```

Comment:
This could be something other than a tuple however. For example, this works fine at runtime:
```python
try:
    raise Exception('foo', 'bar')
except Exception, [a, b]:
    print a, b  # prints foo bar
```
Perhaps just `"Unsupported exception handler syntax ..."`. If you want a specific warning, you can check that the `item.name` is an `ast27.Tuple` or `ast27.List`.

Reply:
I'll change it to `except <expr>, <anything but a name>` -- it's syntactically an assignment target. I still don't think it's worth trying to support other syntax -- we just need to not crash.